src/main/java/io/th0rgal/oraxen/mechanics/provided/misc/food/FoodMechanicListener.java

```diff
@@ -0,0 +1,40 @@
+package io.th0rgal.oraxen.mechanics.provided.misc.food;
+
+import io.th0rgal.oraxen.items.OraxenItems;
+import org.bukkit.GameMode;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerItemConsumeEvent;
+
+public class FoodMechanicListener implements Listener {
+    private FoodMechanicFactory factory;
+
+    public FoodMechanicListener(FoodMechanicFactory factory) {
+        this.factory = factory;
+    }
+
+    @EventHandler
+    public void onEatingFood(PlayerItemConsumeEvent event) {
+        Player player = event.getPlayer();
+        String itemID = OraxenItems.getIdByItem(event.getItem());
+        if (itemID == null || factory.isNotImplementedIn(itemID))
+            return;
+        FoodMechanic mechanic = (FoodMechanic) factory.getMechanic(itemID);
+        event.setCancelled(true);
+        if (player.getGameMode() != GameMode.CREATIVE)
+            player.getInventory().getItemInMainHand().setAmount(player.getInventory().getItemInMainHand().getAmount() - 1);
```

Comment:
add the replacement here. if it is null aka no replacement defined, subtract otherwise give replacement